### PR TITLE
Added delete flag to Dbi.drop which will enable deleting a database.

### PR DESCRIPTION
**Pull request by weidaru**
See https://github.com/lmdbjava/lmdbjava/issues/64 for more details.

### CODE REVIEW

File: src/test/java/org/lmdbjava/DbiTest.java [hunk 1]
```diff
@@ -146,6 +147,22 @@ public void drop() {
   }
 
   @Test
+  public void dropAndDelete() {
+    final Dbi<ByteBuffer> db = env.openDbi(DB_1, MDB_CREATE);
+    final Dbi<ByteBuffer> nameDb = env.openDbi((byte[]) null);
```
Comment (krisskross) [inline src/test/java/org/lmdbjava/DbiTest.java]: What happens if this db is `nameDb.drop(txn, true)`

Comment (weidaru) [inline src/test/java/org/lmdbjava/DbiTest.java]: It will drop the content in nameDb. The MAIN_DB (nameDb) will never get deleted.
https://github.com/LMDB/lmdb/blob/14cff072ec29f48093a9d40cc79934cf5376af4a/libraries/liblmdb/mdb.c#L9949-L9969

File: src/main/java/org/lmdbjava/Dbi.java [hunk 2]
```diff
@@ -186,12 +190,28 @@ public boolean delete(final Txn<T> txn, final T key, final T val) {
...
       txn.checkReady();
       txn.checkWritesAllowed();
     }
-    checkRc(LIB.mdb_drop(txn.pointer(), ptr, 0));
+    if (delete) {
+      clean();
+    }
+    final int del = delete ? 1 : 0;
+    checkRc(LIB.mdb_drop(txn.pointer(), ptr, del));
   }
```
Comment (benalexau) [inline src/main/java/org/lmdbjava/Dbi.java]: I'm unsure I agree that `drop(Txn, true)` should deallocate the comparator keys. The code as shown will deallocate the comparator keys, but this will be reattempted if the user separately invokes `close()`. 

As such either close state needs to be tracked (extra field), the comparator keys need to remove the `final` modifier so we can assign them to `null` (unusual within our `final`-preferring code base), or we simply make it the user's responsibility to invoke `close()` themselves if they wish to release the handles.

While `Txn` offers some precedent for us trying to track state and help the user avoid illegal use, `Txn` is a central API and frequently used. On the other hand, dropping and deleting a DBI is very unusual and I think it reasonable to expect users doing this consider the restrictions it involves (eg making sure no other thread or process is using the DBI, ensuring the now-defunct `Dbi` is no longer accessed once dropped, calling `Dbi.close()` if recovering the small amount of memory concerned out-weights the corruption risks noted in [mdb_dbi_close](http://www.lmdb.tech/doc/group__mdb.html#ga52dd98d0c542378370cd6b712ff961b5) etc).

File: src/test/java/org/lmdbjava/DbiTest.java (hunk 1 above)
Comment (benalexau) [inline src/test/java/org/lmdbjava/DbiTest.java]: Would you mind just adding a quick extra test for `null` database, just so we can see it doesn't segmentation fault on any of the supported platforms (the CI server will confirm this)?

Comment (weidaru) [inline src/test/java/org/lmdbjava/DbiTest.java]: Sure, I can do that.

File: src/main/java/org/lmdbjava/Dbi.java (hunk 2 above)
Comment (weidaru) [inline src/main/java/org/lmdbjava/Dbi.java]: The reason I call clean() is because mdb_drop(del = 1) will close the dbi as well. And I am trying to bring Dbi to a same state as Dbi.close() is called.

Comment (weidaru) [inline src/main/java/org/lmdbjava/Dbi.java]: So the contract is like Dbi.drop(delete = true) calls Dbi.close() for you so you don't need to call it one more time.

Comment (weidaru) [inline src/main/java/org/lmdbjava/Dbi.java]: Make close() idempotent is also a viable approach since lmdb doc also suggest only closing a dbi once. http://www.lmdb.tech/doc/group__mdb.html#gac08cad5b096925642ca359a6d6f0562a

Comment (weidaru) [inline src/main/java/org/lmdbjava/Dbi.java]: I made a revision to have a isClean flag which will protect comparator keys from being deallocated multiple times. User will still need to be aware of handle state when using drop or close.

**Comment by benalexau** [comment]
Thanks @weidaru for this PR and making the changes.
